File: programmers/연습문제/level2/전화번호 목록.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    map<string, int> m;
    
    for (string pb : phone_book) {
        m[pb]++;
    }
    for (int i = 0; i < phone_book.size(); i++) {
        string pb = phone_book[i];
        string tmp = "";
        for (int j = 0; j < pb.length(); j++) {
            tmp += pb[j];
            if (m[tmp] && tmp != pb) {
                return false;
            }
        }
    }
    return answer;
}

File: programmers/연습문제/level2/전화번호 목록.py
def solution(phone_book):
    dic = {}
    
    for pb in phone_book:
        dic[pb] = 1
        
    for pb in phone_book:
        temp = ''
        for ch in pb:
            temp += ch
            if temp in dic and temp != pb:
                return False
    return True